### PR TITLE
Browserify All The Things!

### PR DESCRIPTION
**Pull request by xinan**
This PR is huge, and kind of scary. :boom: 
Please brace yourself. 

Some notes:
* Remove `requirejs` dependency and related codes. 
* Convert the app from `AMD` to `UMD`.
* Move `editor.html` and `frame.html` from `content/src` to `content`.
* Remove `grunt-replace` as it is no longer needed.
* Remove `servesrc` option from server because it is no longer applicable.
* Use `watchify` for `devserver`, `sdevserver` and `testserver` to auto-recompile.
* Remove `filetype.js` from the list of `grunt-watch` because it will be taken care of by `watchify`.

Feel free to ask questions :smile:

### CODE REVIEW

**Comment by ulularem** [comment]
It mostly looks good to me (the things I've reviewed and understood look good). I still need to understand the changes to Gruntfile.js and editor-main.js in more detail.

File: content/src/FontLoader.js [hunk 1]
```diff
@@ -1,591 +1,589 @@
...
-    this._fontFamiliesArray = fontFamiliesArray.slice(0);
-    this._testContainer = null;
-    this._timeoutId = null;
-    this._intervalId = null;
-    this._intervalDelay = 50;
-    this._numberOfLoadedFonts = 0;
-    this._numberOfFontFamilies = this._fontFamiliesArray.length;
-    this._fontsMap = {};
-    this._finished = false;
+var isIE = /MSIE/i.test(navigator.userAgent),
```
Comment (davidbau) [inline content/src/FontLoader.js]: Even though we're not using define(), can you enclose the FontLoader code in a top-level closure `(function() {/*all the code */}).call(this)` or similar?

Comment (davidbau) [inline content/src/FontLoader.js]: Also, if you leave the indent level 2-spaces in, the diff will be cleaner.  :-)

File: Gruntfile.js [hunk 2]
```diff
@@ -333,18 +328,18 @@ module.exports = function(grunt) {
...
-      ['proxymessage', 'express:dev', 'watch']);
+      ['proxymessage', 'express:dev', 'browserify:server', 'watch']);
   // "devserver" serves editor code directly from the src directory.
   grunt.registerTask('sdevserver',
-      ['proxymessage', 'express:sdev', 'watch']);
+      ['proxymessage', 'express:sdev', 'browserify:server', 'watch']);
   // "devserver" serves editor code directly from the src directory.
   grunt.registerTask('testserver',
-      ['proxymessage', 'express:localtest', 'watch']);
+      ['proxymessage', 'express:localtest', 'browserify:server', 'watch']);
```
Comment (dabbler0) [inline Gruntfile.js]: Did you configure `watch` to rebuild the browserify package? Maybe use `watchify` for faster rebuilds if the build is slow.

File: content/src/FontLoader.js (hunk 1 above)
Comment (davidbau) [inline content/src/FontLoader.js]: Oops - I'm new to browserfiy - I guess it probably does this for you.  You can ignore this comment.

File: package.json [hunk 3]
```diff
@@ -30,24 +30,66 @@
...
     "grunt-contrib-watch": "latest",
     "grunt-express-server": "latest",
     "grunt-mocha-test": "latest",
     "grunt-node-inspector": "latest",
-    "grunt-replace": "latest",
     "grunt-sed": "latest",
     "node-phantom-simple": "latest",
-    "phantomjs": "latest"
+    "phantomjs": "latest",
+    "watchify": "latest"
```
Comment (dabbler0) [inline package.json]: Are you using `watchify` (we should if you're not)?

File: Gruntfile.js (hunk 2 above)
Comment (dabbler0) [inline Gruntfile.js]: How is watchify hooked up to the `devserver`, `sdevserver`, and `testserver` tasks?

File: package.json [hunk 4]
```diff
@@ -30,24 +30,66 @@
...
+    "filetype": "./content/src/filetype.js",
+    "gadget": "./content/src/gadget.js",
+    "guide": "./content/src/guide.js",
+    "iced-coffee-script": "./content/lib/iced-coffee-script.js",
+    "jquery": "./content/lib/jquery.js",
+    "palette": "./content/src/palette.js",
+    "pencil-tracer": "./content/lib/pencil-tracer.js",
+    "see": "./content/lib/see.js",
+    "seedrandom": "./content/lib/seedrandom.js",
+    "sourcemap": "./content/lib/sourcemap/source-map-consumer.js",
```
Comment (davidbau) [inline package.json]: Can any of these dependencies (such as sourcemap) be brought in via npm instead of bowercopy?

File: Gruntfile.js (hunk 2 above)
Comment (xinan) [inline Gruntfile.js]: To use `watchify`, simply just set `watch: true` and `keepalive: true` in the `grunt-browserify` options. `devserver`, `sdevserver` and `testserver` use the same `browserify:server` task. Which will watch the source files and do auto-recompile whenever they are modified. 

There is no way to serve source using Browserify. They must be compiled. 

The difference between these 3 servers is that they use different server configs.

Comment (davidbau) [inline Gruntfile.js]: Ah I see.  https://github.com/jmreidy/grunt-browserify#watch

File: package.json (hunk 4 above)
Comment (xinan) [inline package.json]: After some research, seems that `iced-coffee-script`,  `jquery`, `seedrandom` and `source-map` can be brought via `npm`.

**Comment by xinan** [comment]
I've make `uglify` to minimize `editor.js` in the build process to reduce the compiled file from 11MB to 2.2MB. I removed the `pure_getter` option because it is slowing down the minifying process by 4 times and yet the file produced are of almost the same size. 

Also, I removed `browserify:server` task in `testserver` in order to be able to test the minified code.

File: package.json (hunk 4 above)
Comment (davidbau) [inline package.json]: OK - let's go ahead and switch to using npm for these four dependencies for now.  Also, let's switch droplet - @dabbler0, can you "npm publish" droplet?

Comment (davidbau) [inline package.json]: @xinan - before droplet is npm published, will it work to add a package.json dependency using a github URL - https://docs.npmjs.com/files/package.json#git-urls-as-dependencies ?

Comment (sakagg) [inline package.json]: AFAIK droplet is already pushed to npm as `droplet-editor`

**Comment by xinan** [comment]
Please pull it and run it to do more testings :smile: 

(Run `npm install` first. I updated `npm-shrinkwrap.json`, so tell me if it doesn't work on your machine.)

**Comment by davidbau** [comment]
Let's do it!

Merging.

**Pull request merged**